### DISCUSSION TITLE
[General]: Should we transition from CodeSandbox to StackBlitz?

### DISCUSSION THREAD

**Discussion by davidjerleke** [davidjerleke/embla-carousel#722] [General]
### Summary

Hi guys!

At the time of writing, [all examples](https://www.embla-carousel.com/examples/predefined/) on the examples page including the [carousel generator](https://www.embla-carousel.com/examples/generator/) are using the `CodeSandbox` **API** to automatically generate sandboxes.

However, as you might have noticed, `CodeSandbox` will introduce their [20 sandbox limit](https://www.codesandbox.community/c/api-billing-updates/upcoming-pricing-billing-changes) for Free Plans, effective today (30th of January). This might seriously limit collaborative debugging when devs create bug reports and also when devs want to try out multiple examples and realise that they've reached their sandbox limit.

I guess this means that we need to find an alternative. Looking at the [`StackBlitz` pricing](https://stackblitz.com/pricing), the free plan includes **Unlimited projects and collections**, so in my opinion this is a good candidate.

- What do you think? 
- Do you have experience with `StackBlitz`? Pros and cons? 
- Any other alternatives out there? 

Share your thoughts here!

Best,
David

### If applicable, which variants of Embla Carousel are relevant to this conversation?

- [X] embla-carousel-docs **(Documentation)**
- [X] embla-carousel-docs **(Generator)**

**Comment by sarussss** [comment]
Hi @davidjerleke 
It's sad to hear about it, I have no experience using StackBlitz but I think the switch is necessary, hope everything goes well.
Thanks for the work you are doing.

**Comment by Enkratia** [comment]
StackBlitz is much faster than CodeSandbox, so it should improve the project in some way.

**Comment by sarussss** [comment]
Hi @davidjerleke
You can try [Codepen](https://codepen.io/)

**Reply by davidjerleke** [reply]
Thanks for the tip @sarussss.

I think `CodePen` isn't good enough compared to CodeSandbox and StackBlitz. It isn't as intuitive and is missing a lot of features that CodeSandbox and StackBlitz have.

**Comment by davidjerleke** [comment]
I will continue with CodeSandbox for now because they offer unlimited `Dev boxes`.
